image/create.go

```diff
@@ -272,8 +317,16 @@ func GenerateImage(opts ImageCreateOpts) (Image, error) {
 			return Image{}, err
 		}
 
-		ic.PlainSecret = plainSecret
 		ic.CipherSecret = cipherSecret
+
+		if ic.HWKeyIndex < 0 {
+			ic.PlainSecret = plainSecret
```

Comment:
I think the whole block above that starts with `pubKe, err := sec.ParsePubEncKey(pubKeBytes)` should be inside this test, because it is not required when there is a valid `HWKeyIndex` and it should already work with the code you wrote on `Create()` (test `HWKeyIndex` first, so `CipherSecret` can be `nil`). What do you think?

Reply:
you're right.  I moved it.